File: dotnet/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/Factory/Concrete/MacFactory.cs
using AbstractFactoryDesignPattern.Example1.Factory.Abstract;
using AbstractFactoryDesignPattern.Example1.Models.Abstract;
using AbstractFactoryDesignPattern.Example1.Models.Concrete;

namespace AbstractFactoryDesignPattern.Example1.Factory.Concrete;

public class MacFactory : GUIFactory
{
	public override Button CreateButton()
	{
		return new MacButton();
	}
	public override Menu CreateMenu()
	{
		return new MacMenu();
	}
	public override Dialog CreateDialog()
	{
		return new MacDialog();
	}
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/src/factory/concrete/MacFactory.ts
import { GUIFactory } from '../abstract/GUIFactory';
import { Button } from '../../models/abstract/Button';
import { Menu } from '../../models/abstract/Menu';
import { Dialog } from '../../models/abstract/Dialog';
import { MacButton } from '../../models/concrete/MacButton';
import { MacMenu } from '../../models/concrete/MacMenu';
import { MacDialog } from '../../models/concrete/MacDialog';

export class MacFactory extends GUIFactory {
  createButton(): Button { return new MacButton(); }
  createMenu(): Menu { return new MacMenu(); }
  createDialog(): Dialog { return new MacDialog(); }
}
